site/src/main/java/uk/nhs/digital/ps/beans/HippoBeanHelper.java

```diff
@@ -65,4 +67,32 @@ public static String getTaxonomyName() throws HstComponentException {
 
         return taxonomyList;
     }
+
+    /**
+     * Return distinct collection of taxonomy in the format of [Key, Name]
+     */
+    public static List<Entry<String, String>> getTaxonomyKeysAndNames(String[] keys) {
+        List<List<Entry<String, String>>> taxonomyList = new ArrayList<>();
+
+        // For each taxonomy tag key, get the name and also include hierarchy context (ancestors)
+        if (keys != null) {
+            // Lookup Taxonomy Tree
+            TaxonomyManager taxonomyManager = HstServices.getComponentManager().getComponent(TaxonomyManager.class.getName());
+            Taxonomy taxonomyTree = taxonomyManager.getTaxonomies().getTaxonomy(getTaxonomyName());
+
+            for (String key : keys) {
+                List<Category> ancestors = (List<Category>) taxonomyTree.getCategoryByKey(key).getAncestors();
+
+                List<Entry<String, String>> list = ancestors.stream()
+                    .map(category -> new AbstractMap.SimpleEntry<String,String>(category.getKey(),category.getInfo(Locale.UK).getName()))
+                    .collect(Collectors.toList());
```

Comment:
Why don't you just collect to a hashmap rather than a List of Entry's?

Reply:
Will change. That was my original intention :)